Content & UI fixups (#9)
fjahr/asmap-website: content/bitcoin-core.md

layout: default
title: In Bitcoin Core
permalink: /bitcoin-core/
nav_order: 3
---

## ASmap in Bitcoin Core

---------------------

Using an ASmap file in one's Bitcoin Core configuration is already strictly better than the default, and so we want to encourage usage of ASmap among users. However, currently the adoption of the ASmap feature must be assumed to be minimal, since doing so requires creating or procuring an ASMap file and then passing it as an option to the node during startup. Aside from the common wisdom that most users don't change the defaults, there is no trusted way to provide ASmap files for download.

## Usage

#### Create an ASmap with Kartograf

[Kartograf](https://github.com/fjahr/kartograf) is a tool to fetch AS data from regional IRRs, and produce a file to be used with Core. Once we have a reliable process for publishing ASmaps, we'll make them available for public download (tbd).

> #### Compress it with `asmap-tool` in Core

Reviewer: nit: I think it reads better if you write out "Bitcoin Core" each time :)

[asmap-tool](https://github.com/bitcoin/bitcoin/tree/master/contrib/asmap) is a Python script to help encode/compress an ASmap file before passing it to Core.

#### Pass the ASmap to Core

> Core will accept a compressed ASmap file with the `-asmap` option as described [here](https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-0.20.0.md#new-settings).

Reviewer: Hm, I think it's better to point people to the bitcoins help than to older release notes that might be outdated. `src/bitcoind --help`

Reviewer: Also, I think the first two steps could be marked as optional somehow. It's only needed if people want to create a file themselves. But if not they can just go to asmap-data and get the latest file there. We might even want to recommend that for inexperienced users because I am not 100% confident yet that kartograf really is "fool-proof" enough when used by an inexperience user. Say they experience some network disruption during the download phase and miss some files and kartograf doesn't error properly and instead produces an incomplete file. Today I can't guarantee that is impossible and looking at the statistics we print out may not make it obvious enough that something is out of the ordinary.

Author: yea i've clarified this [799cec5](https://github.com/fjahr/asmap-website/pull/9/commits/799cec5e29ae18391a900abbb04ef990c15ff0e5)

## Usage

> `bitcoin` will accept a compressed ASmap file with the `-asmap` option.

Reviewer: ```suggestion
`bitcoind` will accept a compressed ASmap file with the `-asmap` startup option.
```

> You can download a pre-made [latest_asmap.data](https://github.com/fjahr/asmap-data/blob/main/latest_asmap.dat) file in the [asmap-data](https://github.com/fjahr/asmap-data) repo.

Reviewer: minor typo: 'latest_asmap.data'

#### Create an ASmap with Kartograf

> You can choose to generate an ASmap file yourself. [Kartograf](https://github.com/fjahr/kartograf) is a tool to fetch AS data from regional IRRs, and produce a file to be used with Bitcoin Core.

Reviewer: ```suggestion
You can choose to generate an ASmap file yourself. [Kartograf](https://github.com/fjahr/kartograf) is a tool that fetches AS data from multiple sources, combines them, and produces a file with raw map data that can be used in Bitcoin Core after being compressed.
```

Reviewer: I think we should try to find a consistent name for the result file that kartograf produces. It has exactly the same data as the compressed file, yet it's not usable in Bitcoin Core. That may lead to some confusion if we say kartograf produces an asmap file. I usually say something with raw....

#### Compress it with `asmap-tool`

> If you generate a file yourself, you must compress it before passing it to `bitcoind`. [asmap-tool](https://github.com/bitcoin/bitcoin/tree/master/contrib/asmap) is a Python script to help encode/compress an ASmap file which ships with Bitcoin Core.

Reviewer: ```suggestion
If you generate a file yourself, you must compress it before passing it to `bitcoind`. [asmap-tool](https://github.com/bitcoin/bitcoin/tree/master/contrib/asmap) is a Python script to help encode/compress an ASmap file. `asmap-tool` is included in the Bitcoin Core repository.
```

Reviewer: This is really nit picking but 'ships with' sounds to me like it would be usable for someone who downloads a release. But you really need to get the repository.
